### PR TITLE
refactor(a380x/sd): Move mailbox to MSFS avionics framework

### PR DESCRIPTION
**Pull request by flogross89**
## Summary of Changes

Preparation for future development

## Screenshots (if necessary)

image

## Additional context

Discord username (if different from GitHub):

## Testing instructions

Load up airplane, look at SD, should look and behave pretty much exactly as before

## How to download the PR for QA

Every new commit to this PR will cause new A32NX and A380X artifacts to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, find and click on the **PR Build** tab
1. Click on either **flybywire-aircraft-a320-neo** or **flybywire-aircraft-a380-842** download link at the bottom of the page

### CODE REVIEW

**Review by 2hwk — APPROVED** [review]

File: fbw-a380x/src/systems/instruments/src/AtcMailbox/AtcMailbox.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+export class AtcMailbox extends DisplayComponent<AtcMailboxProps> {
+  private readonly topRef = FSComponent.createRef<HTMLDivElement>();
+
+  private readonly mouseCursorRef = FSComponent.createRef<MouseCursor>();
+
+  public onAfterRender(node: VNode): void {
+    super.onAfterRender(node);
+
+    this.topRef.instance.addEventListener('mousemove', (ev) => {
```
Comment (Benjozork) [inline fbw-a380x/src/systems/instruments/src/AtcMailbox/AtcMailbox.tsx:23]: This needs to be cleaned up in `destroy`. You'll need to move the handler to an arrow function in a class property to do that.

**Review by Benjozork — DISMISSED** [review]

File: fbw-a380x/src/systems/instruments/src/AtcMailbox/AtcMailbox.tsx (hunk 1 above)
Comment (flogross89) [reply fbw-a380x/src/systems/instruments/src/AtcMailbox/AtcMailbox.tsx]: Just pushed a commit, like so?

File: fbw-a380x/src/systems/instruments/src/AtcMailbox/AtcMailbox.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+export interface AtcMailboxProps {
+  readonly bus: EventBus;
+}
+
+export class AtcMailbox extends DisplayComponent<AtcMailboxProps> {
+  private readonly topRef = FSComponent.createRef<HTMLDivElement>();
+
+  private readonly mouseCursorRef = FSComponent.createRef<MouseCursor>();
+
+  mouseMoveListener = (ev: MouseEvent) => {
```
Comment (tracernz) [inline fbw-a380x/src/systems/instruments/src/AtcMailbox/AtcMailbox.tsx:20]: Should be private, and probably onMouseMove. Arrows funcs are a little more expensive than .bind'ing a member function too.

Comment (flogross89) [reply fbw-a380x/src/systems/instruments/src/AtcMailbox/AtcMailbox.tsx]: done

**Comment by CaptCrystal** [comment]
Quality Assurance Trainee Report

Discord Username: captcrystal
Object of testing : #9370
Aircraft: A380X
Tier of Testing   : 1/2
Date: 10/11/2024

Testing Process: 

- Loaded in to KSEA
- Verified that the mailbox and all SD pages worked correctly

Testing Results: 
Passed

Conclusions: 
Keep up the great work!

File: fbw-a380x/src/systems/instruments/src/AtcMailbox/AtcMailbox.tsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+  destroy(): void {
+    this.topRef.getOrDefault()?.removeEventListener('mousemove', this.onMouseMoveHandler);
+
+    super.destroy();
+  }
+
+  render(): VNode | null {
+    return (
+      <CdsDisplayUnit bus={this.props.bus} displayUnitId={DisplayUnitID.Sd}>
```
Comment (Benjozork) [inline fbw-a380x/src/systems/instruments/src/AtcMailbox/AtcMailbox.tsx:40]: Hmmm, does this not conflict with the displayunit component of the SD?

Comment (flogross89) [reply fbw-a380x/src/systems/instruments/src/AtcMailbox/AtcMailbox.tsx:40]: In principle yes, but it always shows the same on top of each other so it isn’t an issue. In principle we need a CDS DU for React not showing the overlays, just implementing the power/on/off/dim states

Comment (flogross89) [reply fbw-a380x/src/systems/instruments/src/AtcMailbox/AtcMailbox.tsx:40]: @Benjozork I now disabled the Cds stuff from the underlying React component, only the overlay from the avionics framework is used. I'd appreciate a quick review of that commit, then I can merge

**Review by Benjozork — APPROVED** [review]

**Pull request merged**